api/app/Modules/Library/Http/Controllers/DraftLyricsController.php

```diff
@@ -85,7 +91,17 @@ public function delete(DraftLyrics $draftLyrics): Response
         }
 
         event(new DraftLyricsDeleted($draftLyrics->id));
-        Cache::forget("draftLyricsForTrack:{$draftLyrics->track_id}");
+        $this->unlock($draftLyrics);
+        return response()->noContent();
+    }
+
+    public function publish(DraftLyrics $draftLyrics): Response
+    {
+        if (!Auth::user()->isModerator()) {
+            return response()->noContent();
```

Comment:
Can we use `$this->authorize('publish', $draftLyrics);` like we do elsewhere?

Reply:
Done